chainerx_cc/chainerx/routines/math.cc

```diff
@@ -878,6 +880,88 @@ Array Sqrt(const Array& x) {
+    Array broadcasted_x1 = x1.BroadcastTo(result_shape);
+    Array broadcasted_x2 = x2.BroadcastTo(result_shape);
+
+    Dtype dtype = GetArithmeticResultDtype(x1, x2);
+    Array x1_cast = broadcasted_x1.AsType(dtype);
+    Array x2_cast = broadcasted_x2.AsType(dtype);
+
+    Array out = Empty(x1.shape(), dtype, x1.device());
+    {
+        NoBackpropModeScope scope{};
+        x1_cast.device().backend().CallKernel<PowKernel>(x1_cast, x2_cast, out);
+    }
+
+    {
+        BackwardBuilder bb{"power", {x1_cast, x2_cast}, out};
+        if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+            bt.Define([x1_tok = bb.RetainInput(0), x2_tok = bb.RetainInput(1)](BackwardContext& bctx) {
+                const Array& gout = *bctx.output_grad();
+                const Array& x1 = bctx.GetRetainedInput(x1_tok);
+                const Array& x2 = bctx.GetRetainedInput(x2_tok);
+
+                bctx.input_grad() = gout * x2 * Power(x1, x2 - 1);
+            });
+        }
+        if (BackwardBuilder::Target bt = bb.CreateTarget(1)) {
+            bt.Define([x1_tok = bb.RetainInput(0), out_tok = bb.RetainOutput(0)](BackwardContext& bctx) {
+                const Array& gout = *bctx.output_grad();
+                const Array& x1 = bctx.GetRetainedInput(x1_tok);
+                const Array& out = bctx.GetRetainedOutput(out_tok);
+
+                bctx.input_grad() = gout * out * Log(x1);
+            });
+        }
+
+        bb.Finalize();
+    }
+    return out;
+}
+
+Array Power(const Array& x1, Scalar x2) {
+    Dtype dtype = GetArithmeticResultDtype(x1, x2);
+    Array x1_cast = x1.AsType(dtype);
+    Array out = Empty(x1_cast.shape(), dtype, x1.device());
+    {
+        NoBackpropModeScope scope{};
+        x1.device().backend().CallKernel<PowASKernel>(x1_cast, x2, out);
+    }
+
+    BackwardBuilder bb{"pow_array_scalar", x1_cast, out};
+    if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+        bt.Define([x1_tok = bb.RetainInput(0), x2](BackwardContext& bctx) {
+            const Array& x1 = bctx.GetRetainedInput(x1_tok);
+
+            bctx.input_grad() = *bctx.output_grad() * x2 * Power(x1, x2 + (-1));
+        });
+    }
+    bb.Finalize();
+    return out;
+}
+
+Array Power(Scalar x1, const Array& x2) {
+    Dtype dtype = GetArithmeticResultDtype(x2, x1);
+    Array x2_cast = x2.AsType(dtype);
+    Array out = Empty(x2.shape(), dtype, x2.device());
+    {
+        NoBackpropModeScope scope{};
+        x2.device().backend().CallKernel<PowSAKernel>(x1, x2_cast, out);
+    }
+    BackwardBuilder bb{"pow_scalar_array", x2_cast, out};
+    if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+        bt.Define([out_tok = bb.RetainOutput(0), x1](BackwardContext& bctx) {
+            const Array& out = bctx.GetRetainedOutput(out_tok);
+
+            bctx.input_grad() = *bctx.output_grad() * out * std::log(static_cast<double>(x1));
+        });
+    }
+    bb.Finalize();
+    return out;
+}
```

Comment:
Regarding using `Binary` you can take a look at the latest master as it's has division for scalar / array, i.e. "SA" which should be quite similar to this?

Reply:
Fixed.